### PR TITLE
Clarifying retention settings with regards to segment rolling and various configurations

### PR DESCRIPTION
**Pull request by greg-szrama**
Overhauled language relating to how retention settings work. Clarified the relationship between cluster/topic settings and the segment settings. Fixed language explaining how segment deletion works.

[Link here](https://deploy-preview-116--redpanda-docs-preview.netlify.app/23.3/manage/cluster-maintenance/disk-utilization/)
[Docs issue #2102](https://github.com/redpanda-data/documentation-private/issues/2102)

### CODE REVIEW

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
* A combination of age and aggregate size, triggered when either is exceeded.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
NOTE: Although retention policy is set at a cluster or topic level, it applies to each partition in a topic independently. Within a partition, only closed segments are eligible for deletion or compaction. When a time-based policy is set, all messages within the segment must exceed the set limit. For example, assume the `retention.ms` for a topic is 300,000 ms (5 minutes). If the `segment.ms` is 1,800,000ms (30 minutes) then messages will remain for a minimum of 30 minutes while the segment containing them remains open. In this scenario, as soon as a new segment begins, the retention policy would trigger deletion of the closed segment.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (Feediver1) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
Messages are eligible for deletion after the storage size of the partition containing them exceeds the value specified in `retention_bytes` (the cluster-level property) or `retention.bytes` (the topic-level property). If `retention.bytes` is not set at the topic level, the topic inherits the `retention_bytes` setting. Segments are deleted in chronological order until the partition is back under the specified size.
```

**Review by Feediver1 — APPROVED** [review]
See suggested updates.

**Comment by Feediver1** [comment]
Please update status @greg-szrama . Is this simply awaiting SME review?

**Comment by greg-szrama** [comment]
> Please update status @greg-szrama . Is this simply awaiting SME review?

That's correct. I thought this one was actually already closed out already

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (hcoyote) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc:74]: Might also add a sentence to the end of this like:

> Refer to the Space Management section to see how dynamic storage management can impact local storage cleanup.

Specifically because setting retention_local_strict=false means we can ignore retention.local.target.* settings when the system is under disk pressure, allowing for earlier removal of segmetns.

Comment (hcoyote) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc:74]: (up to y'all though, I know this is already complex and we're threading a needle with how this all works with the if's and when's)

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (hcoyote) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc:79]: Wanted to point out ... there's six total diagrams in the wiki covering how this works, since I had to break it down for each combination of settings to make sure it was understandable. I don't know if we want to make this a simpler diagram, or split them out to talk about each of the conditions?

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (hcoyote) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc:82]: > adjusting your retention properties or enabling dynamic space management are ways to reduce the amount of disk space used

Minor tweak?

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (hcoyote) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Not sure if it's important enough to surface. There's actually a little bit of jitter that comes into play when doing the log segment rolling, so it's not _exactly_ the size of segment.bytes/segment.ms. this is related to the log_segment_ms_min/max clamping. 

Here's the diagram I drew about a few months back when I was figuring out how this worked. 

02b1b76e-fa4d-4278-982d-26253ae0da36

Comment (hcoyote) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: note: while the diagram is written from perspective of time, there's equivalent clamping done on size (log_segment_size_min/max)

**Review by hcoyote — CHANGES_REQUESTED** [review]

**Review by andijcr — APPROVED** [review]

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc:74]: I like this idea, added a sentence with the reference!

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc:79]: If you have a different one in mind, I'm all ears! I chose this one because it was with the docs issue and I thought it presented the information really clearly.

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: I'll try to soften the language around this -- for example, "Segments are closed after they exceed this size..."

**Comment by greg-szrama** [comment]
Suggested updates are complete, ready for review and merge.

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
By default, all topics on self-hosted Redpanda clusters retain 24 hours of data on local disk, while Redpanda Cloud retains 12 hours of data. If data is written fast enough to consume all disk space within the local retention time period, it's possible to exhaust local disk space, even when using Tiered Storage. You can adjust the default local storage size or time period, if required.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
* A combination of message age and aggregate size, triggered when either is exceeded.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
NOTE: Although retention policy is set at a cluster or topic level, it applies to each partition in a topic independently. Within a partition, only closed segments are eligible for deletion or compaction. When a time-based policy is set, all messages within the segment must exceed the set limit. For example, assume the `retention.ms` for a topic is 300,000 ms (5 minutes). If the `segment.ms` is 1,800,000 ms (30 minutes), then messages remain for a minimum of 30 minutes while the segment containing them remains open. In this scenario, as soon as a new segment begins, the retention policy would trigger deletion of the closed segment.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
Retention policy functions by deleting or compacting closed segments. The segment lifetime and segment size configurations help ensure new segments are created regularly within each partition. This illustration shows the relationship between segments and the retention policy using time-based limits. If these values are set too high, especially if they're higher than the corresponding retention values, the segment may fill available disk space before closing and therefore never become eligible for deletion or compaction. If the values are set too low, your partitions will have a large number of segments that must be checked each time deletion or compaction processes execute, having a potential adverse impact on system resource utilization.
```
I was taught for accessibility (screen readers) never to use navigations in doc like above/below.

Comment (micheleRP) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Can you get Ben Weinstein to make this image consistent with our image standards? I have trouble understanding what it's trying to show. And I hate the split case "Message & Log segment rolling based on Time"

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
Messages are eligible for deletion after the storage size of the partition containing them exceeds the value specified in `retention_bytes` (the cluster-level property) or `retention.bytes` (the topic-level property). If `retention.bytes` is not set at the topic level, the topic inherits the `retention_bytes` setting. Segments are deleted in chronological order until the partition is back under the specified size limit.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
The `log_segment_size` property specifies the size of each log segment within the partition. Segments are closed after they reach this size, and messages begin filling a new segment.
```

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (micheleRP) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]:
```suggestion
Writing data for a topic usually spans multiple log segments. An *active segment* of a topic is a log segment that is being written to. As data of a topic is written and an active segment becomes full (reaches `log_segment_size`), it's closed and changed to read-only mode. A new segment is created and set to read-write mode, and it becomes the active segment. *Log rolling* is the rotation between segments to create a new active segment.
```

**Review by micheleRP — APPROVED** [review]
approving with some suggested edits

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: I snagged the image from Travis, let me see if we can get it tidied up!

Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Image replaced and language revised to focus on how segments roll and how that impacts retention.

**Comment by greg-szrama** [comment]
@hcoyote Can you give this a once-over? I've resolved the suggestions you left. Thanks!

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (mattschumpert) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Our new cloud tiers are sized for 6 hours of retention but use Space Management to dynamically manage storage.

Comment (mattschumpert) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Also with Space Management enabled its much less likely to run out of local space because we purge data in stages to avoid this. Perhaps we should refer them to the Space Management doc topic here.

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (mattschumpert) [inline modules/manage/pages/cluster-maintenance/disk-utilization.adoc:93]: We should also point out this once it's documented. (the fact that errant CreateTime timestamps can be corrected automatically by Redpanda: https://github.com/redpanda-data/documentation-private/issues/1980)

**Review by mattschumpert — APPROVED** [review]

**Review by hcoyote — APPROVED** [review]

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc:93]: Yessir! I have that docs issue as well and will be amending this language as soon as we merge this PR.

File: modules/manage/pages/cluster-maintenance/disk-utilization.adoc
Comment (greg-szrama) [reply modules/manage/pages/cluster-maintenance/disk-utilization.adoc]: Updated the language:

By default, all topics on self-hosted Redpanda clusters retain 24 hours of data on local disk while Redpanda Cloud retains 6 hours of data. On Redpanda cloud, [space management](https://github.com/redpanda-data/docs/pull/116#space_management) is automatically enabled to dynamically manage storage and you can also enable this on self-hosted clusters. Even so, if data is written fast enough to consume all disk space within the local retention time period, it’s possible to exhaust local disk space, even when using Tiered Storage. You can adjust the default local storage size or time period, if required.

**Pull request merged**
